render.go

```diff
@@ -171,44 +165,104 @@ func (r *Render) Render(buffer *Buffer, completion *CompletionManager) {
 	prefix := r.getCurrentPrefix()
 
 	// Erasing
-	r.out.CursorBackward(int(r.col) + len(buffer.Text()) + len(prefix))
-	r.out.EraseDown()
+	r.clear(r.previousCursor)
 
 	// prepare area
 	line := buffer.Text()
-	h := ((len(prefix) + len(line)) / int(r.col)) + 1 + int(completion.max)
+	cursor := stringWidth(prefix) + stringWidth(line)
+	_, y := r.toPos(cursor)
+
+	h := y + 1 + int(completion.max)
 	if h > int(r.row) || completionMargin > int(r.col) {
 		r.renderWindowTooSmall()
 		return
 	}
 
 	// Rendering
 	r.renderPrefix()
-
 	r.out.SetColor(r.inputTextColor, r.inputBGColor, false)
 	r.out.WriteStr(line)
 	r.out.SetColor(DefaultColor, DefaultColor, false)
-	r.out.CursorBackward(len([]rune(line)) - buffer.CursorPosition)
+
+	cursor = r.backward(cursor, stringWidth(line)-buffer.CursorPosition)
+
 	r.renderCompletion(buffer, completion)
 	if suggest, ok := completion.GetSelectedSuggestion(); ok {
-		r.out.CursorBackward(len([]rune(buffer.Document().GetWordBeforeCursor())))
+		cursor = r.backward(cursor, stringWidth(buffer.Document().GetWordBeforeCursor()))
+
 		r.out.SetColor(r.previewSuggestionTextColor, r.previewSuggestionBGColor, false)
 		r.out.WriteStr(suggest.Text)
 		r.out.SetColor(DefaultColor, DefaultColor, false)
+
+		cursor += stringWidth(suggest.Text)
 	}
 	r.out.Flush()
+
+	r.previousCursor = cursor
 }
 
 // BreakLine to break line.
 func (r *Render) BreakLine(buffer *Buffer) {
-	// CR
-	prefix := r.getCurrentPrefix()
-	r.out.CursorBackward(int(r.col) + len(buffer.Text()) + len(prefix))
 	// Erasing and Render
-	r.out.EraseDown()
+	r.clearLine(buffer)
 	r.renderPrefix()
 	r.out.SetColor(r.inputTextColor, r.inputBGColor, false)
 	r.out.WriteStr(buffer.Document().Text + "\n")
 	r.out.SetColor(DefaultColor, DefaultColor, false)
 	r.out.Flush()
+
+	r.previousCursor = 0
+}
+
+func (r *Render) clearLine(buffer *Buffer) {
```

Comment:
I imagine clearLine means just erase line but this method erase from the current line until the bottom of terminal. we need to consider the naming.